elements/lisk-p2p/test/unit/peer/base.ts

```diff
@@ -399,24 +391,17 @@ describe('peer/base', () => {
 				sandbox.stub(defaultPeer, 'request').throws('Error');
 			});
 
-			it(`should emit ${EVENT_FAILED_TO_FETCH_PEERS} event`, async () => {
-				try {
-					await defaultPeer.fetchPeers();
-				} catch (e) {
-					expect(defaultPeer.emit).to.be.calledOnceWith(
-						EVENT_FAILED_TO_FETCH_PEERS,
-					);
-				}
+			it(`should emit ${EVENT_FAILED_TO_FETCH_PEERS} event`, () => {
+				expect(defaultPeer.fetchPeers()).to.be.rejected;
+				expect(defaultPeer.emit).to.be.calledOnceWith(
+					EVENT_FAILED_TO_FETCH_PEERS,
+				);
 			});
 
-			it('should throw an error', async () => {
-				try {
-					await defaultPeer.fetchPeers();
-				} catch (e) {
-					expect(e).to.be.an.instanceOf(RPCResponseError);
-					expect(e.message).to.be.eql('Failed to fetch peer list of peer');
-					expect(e.peerId).to.be.eql(defaultPeerInfo.ipAddress);
-				}
+			it('should throw an error', () => {
```

Comment:
Thank you for pointing out. I found out we need `return` when using `.to.be.rejectedWith`, otherwise if it fails we have an unhandled rejection. However, I don't see the need of `async`.